ui/e2e/consumers/index.spec.ts

```diff
@@ -0,0 +1,203 @@
+import { createNamespace, deleteNamespace } from "../utils/namespace";
+import { createRedisConsumerFile, findConsumerWithApiRequest } from "./utils";
+import { expect, test } from "@playwright/test";
+
+import { createWorkflow } from "~/api/tree/mutate/createWorkflow";
+import { headers } from "e2e/utils/testutils";
+import { updateWorkflow } from "~/api/tree/mutate/updateWorkflow";
+
+let namespace = "";
+
+test.beforeEach(async () => {
+  namespace = await createNamespace();
+});
+
+test.afterEach(async () => {
+  await deleteNamespace(namespace);
+  namespace = "";
+});
+
+test("Consumer list is empty by default", async ({ page }) => {
+  await page.goto(`/${namespace}/gateway/consumers`, {
+    waitUntil: "networkidle",
+  });
+
+  await expect(
+    page.getByTestId("breadcrumb-consumers"),
+    "it renders the 'Consumers' breadcrumb"
+  ).toBeVisible();
+
+  await expect(
+    page.getByText("No consumers exist yet"),
+    "it renders an empty list of consumers"
+  ).toBeVisible();
+});
+
+test("Consumer list shows all available consumers", async ({ page }) => {
+  await createWorkflow({
+    payload: createRedisConsumerFile({
+      username: "userA",
+      password: "password",
+    }),
+    urlParams: {
+      baseUrl: process.env.VITE_DEV_API_DOMAIN,
+      namespace,
+      name: "redis-consumer.yaml",
+    },
+    headers,
+  });
+
+  await expect
+    .poll(
+      async () =>
+        await findConsumerWithApiRequest({
+          namespace,
+          match: (consumer) => consumer.username === "userA",
+        }),
+      "the consumer was created and is available"
+    )
+    .toBeTruthy();
+
+  await page.goto(`/${namespace}/gateway/consumers`, {
+    waitUntil: "networkidle",
+  });
+
+  await expect(
+    page.getByTestId("consumer-table").locator("tr"),
+    "it renders one row of consumers"
+  ).toHaveCount(1);
+
+  await expect(
+    page.getByTestId("consumer-table").getByRole("cell").nth(0),
+    "it renders the field for the username"
+  ).toBeVisible();
```

Comment:
For simplicity, we could even remove this assertion, because it is covered in the upcoming assertion anyway. Even if it does not do any harm to leave this here, the description "it renders the field for the username" is not 100 % accurate, because we have nothing in the selector that tells us anything about the username, we just check if there is a first column.

Reply:
Agreed, I was beginning to question these tests myself. It makes sense to only use as many tests as are needed.